rfcs/20221006-frontend_event_tracking.md

```diff
@@ -0,0 +1,296 @@
+  APPLY_FILTER: {
+    /** The name of the filter (not the user-facing element label) */
+    key: string;
+    /** The value of the filter */
+    value: string | boolean;
+    /** The media type being searched */
+    mediaType: MediaType;
+    /** The search term **/
+    query: string;
+  };
+  /** Whenever a user changes the content type */
+  CHANGE_CONTENT_TYPE: {
+    /** The previously-set media type */
+    previous: MediaType;
+    /** The new media type */
+    next: MediaType;
+    /** The name of the Vue component used to switch content types. **/
+    component: string;
+  };
+  /**
+   * Whenever the user scrolls to the end of the results page.
+   * Useful to evaluate how often users load more results or click
+   * on the external sources dropdown.
+   * */
+  REACH_RESULT_END: {
+    /** The media type being searched */
+    mediaType: MediaType;
+    /** The search term **/
+    query: string;
+  };
+  /** Whenever the user clicks the load more button */
+  LOAD_MORE_RESULTS: {
+    /** The media type being searched */
+    mediaType: MediaType;
+    /** The search term **/
+    query: string;
+  };
+  /** When a user opens the external sources popover. */
+  VIEW_EXTERNAL_SOURCES: {
+    /** The media type being searched */
+    mediaType: MediaType;
+    /** The search term **/
+    query: string;
+  };
+  /** When the user chooses an external source from the dropdown of external sources */
+  SELECT_EXTERNAL_SOURCE: {
+    /** The name of the external source */
+    name: string;
+    /** The full URL of the source */
+    link: string;
+    /** The media type being searched */
+    mediaType: MediaType;
+    /** The search term **/
+    query: string;
+  };
+  /**  The user opens the menu which lists pages. */
+  OPEN_PAGES_MENU: {};
+  /** Whenever the user selects a result from the search results page. */
+  SELECT_SEARCH_RESULT: {
+    /** If the result is a related result, provide the ID of the 'original' result */
+    related: string | null;
+    /** The media type being searched */
+    mediaType: MediaType;
+    /** The slug (not the prettified name) of the provider */
+    provider: string;
+    /** The search term **/
+    query: string;
+  };
+  /** The user clicks the CTA button to the external source to use the image */
+  GET_MEDIA: {
+    /** The unique ID of the media */
+    id: string;
+    /** The slug (not the prettified name) of the provider */
+    provider: string;
+    /** The media type being searched */
+    mediaType: MediaType;
+  };
+  /** The user clicks one of the buttons to copy the media attribution */
+  COPY_ATTRIBUTION: {
+    /** The unique ID of the media */
+    id: string;
+    /** The format of the copied attribution */
+    format: "plain" | "rich" | "html";
+    /** The media type being searched */
+    mediaType: MediaType;
+  };
+  /** The user reports a piece of media through our form */
+  REPORT_MEDIA: {
+    /** The unique ID of the media */
```

Comment:
> This actually gives me an idea, we may want to add the version number of Openverse to the events as another default field.

Doesn't sound like a bad idea generally.

> Is this event necessary since we are already receiving the report?

@krysal is on to something here that we should not quickly dismissed. I hadn't considered it earlier, but we should consider that long-term analysis of our usage would probably logically include our own data in addition to the events themselves.

However, the reports don't record who did it or if they're done in the same session, so this kind of thing might be useful for identifying if there are specific "power users" or "power sessions" where people go and report lots of things all at once. Not sure what we would do with that information though :slightly_smiling_face: 

If the question is "how many do we receive on average" etc, then we can set up a Grafana dashboard eventually to pull that from the database itself, rather than denormalising the data into an event stream separate from the actual data itself.

This is more of a general principle/consideration though. I don't know how/if it applies specifically to the events here. Something to consider for these and future events though. We want to have a wholistic approach to our analytics that incorporates all the data streams we have, not just user event data.